src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -6801,6 +6801,12 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_CollectionLiteralNoTargetType" xml:space="preserve">
     <value>There is no target type for the collection literal.</value>
   </data>
+  <data name="ERR_CollectionBuilderAttributeMethodNotFound" xml:space="preserve">
+    <value>Could not find an accessible '{0}' method with the expected signature.</value>
+  </data>
+  <data name="ERR_CollectionBuilderNoElementType" xml:space="preserve">
+    <value>'{0}' has a CollectionBuilderAttribute but no element type.</value>
```

Comment:
When does this happen? When the collection type lacks the normal cues for its element type such as `GetEnumerator().Current`?

Reply:
Yes, this is for collection types that are not "foreach-able". That is, collections that do not implement `IEnumerable<T>` or `IEnumerable` and do not implement the enumerable pattern.